client/asset/eth/eth.go

```diff
@@ -1060,16 +1060,33 @@ func (eth *ExchangeWallet) SwapConfirmations(ctx context.Context, _ dex.Bytes, c
 	return
 }
 
-// Withdraw withdraws funds to the specified address. Value is gwei.
-//
-// TODO: Return the asset.Coin.
-// TODO: Subtract fees from the value.
+// Withdraw withdraws funds to the specified address. Value is gwei. The fee is
+// subtracted from the total balance if it cannot be sent otherwise.
 func (eth *ExchangeWallet) Withdraw(addr string, value, _ uint64) (asset.Coin, error) {
-	_, err := eth.node.sendToAddr(eth.ctx, common.HexToAddress(addr), value)
+	eth.lockedFundsMtx.Lock()
+	defer eth.lockedFundsMtx.Unlock()
```

Comment:
Can these be RLock/RUnlock?

Reply:
The act of sending changes what is returned by `balance` so I think holding the write lock is optimal as it changes the amount locked.